Add Documentation for deploying Antrea with Rancher cluster (#4733)
antrea-io/antrea: docs/rancher.md

# Deploying Antrea on Rancher

<!-- toc -->
- [Deploy Rancher Server](#deploy-rancher-server)
- [Deploy Antrea](#deploy-antrea)
<!-- /toc -->

## Deploy Rancher Server

> Follow these [steps](https://minikube.sigs.k8s.io/docs/start) to deploy rancher server ui dashboard.

Reviewer: this link is about minikube,
https://minikube.sigs.k8s.io/docs/start/

Reviewer: Needs to elaborate the steps deploying rancher server or provide right link

> ## Deploy Antrea

Reviewer: Better use "Deploy Antrea using Rancher UI" instead of "Deploy Antrea"

* Sign-in to the UI dashboard, and switch to cluster management.

> * Create new node template there under the RKE configuration section.

Reviewer 2: node -> Node

> * After saving the template, proceed with creating new cluster, under the clusters tab click on create button, then enter the name, and number of worker and controlplane nodes respectively.

Reviewer 2: nodes -> Nodes

> * click on “Edit as YAML’, and set the network plugin to null, and add a addon for antrea, in the following manner:

Reviewer 2: ```suggestion
* Click on "Edit as YAML", and set the network plugin to null, and add an addon for antrea, in the following manner:
```

# Deploying Antrea on Rancher Managed Cluster

<!-- toc -->
- [Deploy Rancher Server](#deploy-rancher-server)
- [Deploy Antrea Using Rancher UI](#deploy-antrea-using-rancher-ui)
<!-- /toc -->

## Deploy Rancher Server

> 

Reviewer: We should add version information with which rancher and antrea was tried to remove ambiguity, there might be many queries raised if the previous version did not work.

Follow these [steps](https://rancher.com/docs/rancher/v1.6/en/installing-rancher/installing-server/) to deploy rancher server ui dashboard.

## Deploy Antrea Using Rancher UI

* Sign-in to the UI dashboard, and switch to cluster management.
* Create new Node template there under the RKE configuration section.
* After saving the template, proceed with creating new cluster, under the clusters tab click on create button, then enter the name, and number of worker and controlplane Nodes respectively.
* click on “Edit as YAML", and set the network plugin to null, and add a addon for antrea, in the following manner:
addons_include:
  -<link of the antrea.yml file>
* You can find few screenshots for reference:
![rancher_cluster](https://raw.githubusercontent.com/antrea-io/antrea/main/docs/assets/rancher_cluster.png)
![rancher_create](https://raw.githubusercontent.com/antrea-io/antrea/main/docs/assets/rancher_create.png)

> ![rancher_node](https://raw.githubusercontent.com/antrea-io/antrea/main/docs/assets/rancher_node.png)

Reviewer: Screenshot should be made available at the same place where step is followed

> # Deploying Antrea on Rancher Managed Cluster

Reviewer 3: I still have the same comment as last time. I believe it would make more sense to move all this information to the Rancher section in docs/kubernetes-installers.md. I am not sure this warrants a standalone document.

## Deploy Rancher Server UI Dashboard

Follow these [steps](https://rancher.com/docs/rancher/v1.6/en/installing-rancher/installing-server/) to deploy rancher server ui dashboard.

## Deploy Antrea Using Rancher UI

> * Sign-in to the UI dashboard, and switch to cluster management.

Reviewer 4: ```suggestion
* Sign in to the UI dashboard, and switch to cluster management.
```

<img src="https://downloads.antrea.io/static/04062023/rancher_cluster_management.png" width="600">

* Create new Node template there under the RKE configuration section.

<img src="https://downloads.antrea.io/static/04062023/rancher_node_template.png" width="600">

> * After saving the template, proceed with creating new cluster, under the clusters tab click on create button, then enter the name, and number of worker and controlplane Nodes respectively.

Reviewer 4: ```suggestion
* After saving the template, proceed with creating new cluster, click on create button under the clusters tab, then enter the name, number of worker and control plane Nodes respectively.
```

> <img src="https://downloads.antrea.io/static/04062023/rancher_create.png" width="600">

Reviewer 4: I feel the width can be 800 or 1000, please check and make it clearer with default width.

Author: okay i will make it 800, but in some docs the width is 600 as well.

# Deploying Antrea on Rancher Managed Cluster

<!-- toc -->
- [Deploy Rancher Server](#deploy-rancher-server)
- [Deploy Antrea Using Rancher UI](#deploy-antrea-using-rancher-ui)

> <!-- /toc -->

Reviewer 3: I feel like including a TOC for such a small document is not necessary. You could add the file to `hack/.notableofcontents` instead.

## Deploy Rancher Server

> Follow these [steps](https://rancher.com/docs/rancher/v1.6/en/installing-rancher/installing-server/) to deploy rancher server ui dashboard.

Reviewer 3: deploy the Rancher server UI dashboard

Reviewer 4: @jainpulkit22 this is not addressed.

## Deploy Antrea Using Rancher UI

* Sign-in to the UI dashboard, and switch to cluster management.

![rancher_cluster_management](https://raw.githubusercontent.com/antrea-io/antrea/main/docs/assets/rancher_cluster_management.png)

* Create new Node template there under the RKE configuration section.

![rancher_node_template](https://raw.githubusercontent.com/antrea-io/antrea/main/docs/assets/rancher_node_template.png)

* After saving the template, proceed with creating new cluster, under the clusters tab click on create button, then enter the name, and number of worker and controlplane Nodes respectively.

![rancher_create](https://raw.githubusercontent.com/antrea-io/antrea/main/docs/assets/rancher_create.png)

* click on “Edit as YAML", and set the network plugin to null, and add a addon for antrea, in the following manner:
addons_include:

>   -<link of the antrea.yml file>

Author: here the user can use any yml file, I have not specified the exact link here because then the user might be of the thought that we need to use this particular yml only, that's why I have mentioned it like this to make it more general.

> You can find few screenshots for validation after following the above steps:

Reviewer 4: ```suggestion
You can find a few screenshots for validation after following the above steps:
```

![rancher_cluster](https://raw.githubusercontent.com/antrea-io/antrea/main/docs/assets/rancher_cluster.png)
![rancher_node](https://raw.githubusercontent.com/antrea-io/antrea/main/docs/assets/rancher_node.png)

> Note: Antrea(version v1.10+) is supported on rancher version v2.7.0+.

Reviewer 4: ```suggestion
Note: Antrea (>= v1.10.0) is supported on Rancher version v2.7.0+.
```
or `(since v1.10.0)`
